RHIDP-5604 Configuring entity tab titles and entity detail tab layout (#903)
redhat-developer/red-hat-developers-documentation-rhdh: modules/customizing-the-appearance/proc-customizing-entity-tab-titles.adoc

[id="configuring-entity-tab-titles_{context}"]
= Configuring entity tab titles

{product} provides an default opinionated tab set for catalog entity views.
For consistency with your organization needs, you can rename, reorder, remove, and add tab titles.

.Procedure

. To rename the default tabs, add the following `entityTabs` section in your `{my-app-config-file}` file, and change the `title` values.
+
[source,yaml]
----
upstream:
  backstage:
    appConfig:
      dynamicPlugins:
        frontend:

>           entity-tabs-customization:

Reviewer: This should be the plugin name.
Check the comments section here https://issues.redhat.com/browse/RHDHBUGS-161

Reviewer: you can add the `entityTabs` property to the plugin configuration to rename/reorder the entity tabs, provided the plugin is enabled

Author: What would be the accurate plugin name in this context? 

Note: the example is taken from https://docs.google.com/document/d/1hPS8nHa1zAu_BpVIIr1KNcbt55PMQ8tqXJWOY3Hs0tg/edit?tab=t.0#heading=h.b084i6kjgg9r and it is an extended version from the example hre: https://issues.redhat.com/browse/RHDHBUGS-161?focusedId=26707948&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-26707948.

Author: Oh I see the later comment https://issues.redhat.com/browse/RHDHBUGS-161?focusedId=26733952&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-26733952.

Should I use `backstage-community.plugin-topology` for the plugin name? 

```suggestion
          backstage-community.plugin-topology:
```

>           entity-tabs-customization:
>             entityTabs:
>               - path: /
>                 title: Overview
>                 mountPoint: entity.page.overview
>               - path: /topology
>                 title: Topology
>                 mountPoint: entity.page.topology
>               - path: /issues
>                 title: Issues
>                 mountPoint: entity.page.issues
>               - path: /pr
>                 title: Pull/Merge Requests
>                 mountPoint: entity.page.pull-requests
>               - path: /ci
>                 title: CI
>                 mountPoint: entity.page.ci
>               - path: /cd
>                 title: CD
>                 mountPoint: entity.page.cd
>               - path: /kubernetes
>                 title: Kubernetes
>                 mountPoint: entity.page.kubernetes
>               - path: /image-registry
>                 title: Image Registry
>                 mountPoint: entity.page.image-registry
>               - path: /monitoring
>                 title: Monitoring
>                 mountPoint: entity.page.monitoring
>               - path: /lighthouse
>                 title: Lighthouse
>                 mountPoint: entity.page.lighthouse
>               - path: /api
>                 title: Api
>                 mountPoint: entity.page.api
>               - path: /dependencies
>                 title: Dependencies
>                 mountPoint: entity.page.dependencies
>               - path: /docs
>                 title: Docs
>                 mountPoint: entity.page.docs
>               - path: /definition
>                 title: Definition
>                 mountPoint: entity.page.definition
>               - path: /system
>                 title: Diagram
>                 mountPoint: entity.page.diagram

Author: ```suggestion
          backstage-community.plugin-topology:
            entityTabs:
              - mountPoint: entity.page.topology
                path: /topology
                title: Changed Topology
              - mountPoint: entity.page.lighthouse
                path: /lighthouse
                title: Changed Lighthouse
              - mountPoint: entity.page.api
                path: /api
                title: Changed Api
              - mountPoint: entity.page.dependencies
                path: /dependencies
                title: Changed Dependencies
              - mountPoint: entity.page.docs
                path: /docs
                title: Changed Docs
              - mountPoint: entity.page.overview
                path: /
                title: Changed Overview
          backstage.plugin-kubernetes:
            entityTabs:
              - mountPoint: entity.page.kubernetes
                path: /kubernetes
                title: Changed Kubernetes
```

[id="configuring-entity-tab-titles_{context}"]
= Configuring entity tab titles

> {product} provides an default opinionated tab set for catalog entity views.

Author: ```suggestion
{product} provides a default opinionated tab set for catalog entity views.
```

For consistency with your organization needs, you can rename, reorder, remove, and add tab titles.

.Procedure
* For each tab to modify, enter your desired values in the `entityTabs` section in your `{my-app-config-file}` file:
+
[source,yaml,subs="+quotes"]
----
upstream:
  backstage:
    appConfig:
      dynamicPlugins:

>         frontend:
>          _<your_plugin_name>_:
>             entityTabs:
>               - mountPoint: _<mount_point>_
>                 path: _<path>_
>                 title: _<title>_
>                 priority: _<priority>_
> ----
> 
> `_<your_plugin_name>_`::
> Enter the plugin name, such as `backstage-community.plugin-topology`.

Author: ```suggestion
        frontend:
         _<plugin_name>_:
            entityTabs:
              - mountPoint: _<mount_point>_
                path: _<path>_
                title: _<title>_
                priority: _<priority>_
----

`_<plugin_name>_`::
Enter the plugin name, such as `backstage-community.plugin-topology`.
```

+
[source,yaml]
----
upstream:
  backstage:
    appConfig:
      dynamicPlugins:
        frontend:
          backstage-community.plugin-topology:
            entityTabs:
              - mountPoint: entity.page.topology
                path: /topology
                title: Changed Topology
              - mountPoint: entity.page.lighthouse
                path: /lighthouse
                title: Changed Lighthouse
              - mountPoint: entity.page.api
                path: /api
                title: Changed Api
              - mountPoint: entity.page.dependencies
                path: /dependencies
                title: Changed Dependencies
              - mountPoint: entity.page.docs
                path: /docs
                title: Changed Docs
              - mountPoint: entity.page.overview
                path: /
                title: Changed Overview
          backstage.plugin-kubernetes:
            entityTabs:
              - mountPoint: entity.page.kubernetes
                path: /kubernetes
                title: Changed Kubernetes
----

> . To reorder tabs, set the `priority` value. Higher priority appears first.

Author: ```suggestion
. To reorder tabs, set the `priority` value. Higher priority appears first.
For example to set the *Topology* tab priority to `42`:
```

+
[source,yaml,subs="+quotes"]
----
upstream:
  backstage:
    appConfig:
      dynamicPlugins:
        frontend:

>           entity-tabs-customization:
>             entityTabs:
>               - path: _<tab_path>_
>                 title: _<tab_title>_
>                 mountPoint: _<tab_mount_point>_
>                 priority: 42

Author: ```suggestion
          backstage-community.plugin-topology:
            entityTabs:
              - mountPoint: entity.page.topology
                path: /topology
                title: Topology
                priority: 42
```

              - mountPoint: entity.page.api
                path: /api
                title: Changed Api
              - mountPoint: entity.page.dependencies
                path: /dependencies
                title: Changed Dependencies
              - mountPoint: entity.page.docs
                path: /docs
                title: Changed Docs
              - mountPoint: entity.page.overview
                path: /
                title: Changed Overview
          backstage.plugin-kubernetes:
            entityTabs:
              - mountPoint: entity.page.kubernetes
                path: /kubernetes
                title: Changed Kubernetes
----

. To reorder tabs, set the `priority` value. Higher priority appears first.
For example to set the *Topology* tab priority to `42`:
+
[source,yaml,subs="+quotes"]
----
upstream:
  backstage:
    appConfig:
      dynamicPlugins:
        frontend:
          backstage-community.plugin-topology:
            entityTabs:
              - mountPoint: entity.page.topology
                path: /topology
                title: Topology
                priority: 42
----

. To remove a tab, set the tab `priority` to a negative value:

> +
> [source,yaml,subs="+quotes"]
> ----
> upstream:
>   backstage:
>     appConfig:
>       dynamicPlugins:
>         frontend:
>           entity-tabs-customization:
>             entityTabs:
>               - path: _<tab_path>_
>                 title: _<tab_title>_

Author: ```suggestion
For example to remove the *Topology* tab:
+
[source,yaml,subs="+quotes"]
----
upstream:
  backstage:
    appConfig:
      dynamicPlugins:
        frontend:
          backstage-community.plugin-topology:
            entityTabs:
              - mountPoint: entity.page.topology
                path: /topology
                title: Topology
```

[id="configuring-entity-tab-titles_{context}"]
= Configuring entity tab titles

{product} provides an default opinionated tab set for catalog entity views.
For consistency with your organization needs, you can rename, reorder, remove, and add tab titles.

.Procedure

. To rename the default tabs, add the following `entityTabs` section in your `{my-app-config-file}` file, and change the `title` values.
+
[source,yaml]
----
upstream:
  backstage:
    appConfig:
      dynamicPlugins:
        frontend:

>           backstage-community.plugin-topology:

Reviewer: You could mention how you did in the other files
```suggestion
          _<your_plugin_name>_:  # same as `scalprum.name` key in plugin's `package.json`
```

Donot want the customers to get an impression that they can do this only with topology/k8s plugin

Reviewer: I think you can use this same example that you have used above

```
global:
  dynamic:
    plugins:
      - package: _<your_package_location>_ # <1>
        disabled: false
        pluginConfig:
          dynamicPlugins:
            frontend:
              _<your_plugin_name>_: # <2>
                mountPoints:
                  - mountPoint: entity.page.ci/cards # <3>
                    importName: TektonCI # <4>
                    config:
                      layout: # <5>
                        gridColumn:
                          lg: span 6
                          xs: span 12
                      if:
                        allOf:
                          - isTektonCIAvailable
 ```
 
 
 ```
 global:
  dynamic:
    plugins:
      - package: _<your_package_location>_ # <1>
        disabled: false
        pluginConfig:
          dynamicPlugins:
            frontend:
              _<your_plugin_name>_: # <2>
                  entityTabs:
                     - mountPoint: entity.page.topology
                        path: /topology
                        title: Changed Topology
                     - mountPoint: entity.page.lighthouse
                        path: /lighthouse
                        title: Changed Lighthouse
                     - mountPoint: entity.page.api
                        path: /api
                        title: Changed Api
                     - mountPoint: entity.page.dependencies
                        path: /dependencies
                        title: Changed Dependencies
                     - mountPoint: entity.page.docs
                        path: /docs
                        title: Changed Docs
                     - mountPoint: entity.page.overview
                        path: /
                        title: Changed Overview
 ```

Author: Yes, but what are the expected values for _<your_package_location>_ and _<your_plugin_name>_?

* The user can decide any random value?
* The user must use the correct package location and plugin name? Clarify either how to find the information, if the values might change, or the expected value, if the values are consistent and reproducible.

                title: Changed Overview
          backstage.plugin-kubernetes:
            entityTabs:
              - mountPoint: entity.page.kubernetes
                path: /kubernetes
                title: Changed Kubernetes
----

. To reorder tabs, set the `priority` value. Higher priority appears first.
For example to set the *Topology* tab priority to `42`:
+
[source,yaml,subs="+quotes"]
----
upstream:
  backstage:
    appConfig:
      dynamicPlugins:
        frontend:
          backstage-community.plugin-topology:
            entityTabs:
              - mountPoint: entity.page.topology
                path: /topology
                title: Topology
                priority: 42
----

. To remove a tab, set the tab `priority` to a negative value:
For example to remove the *Topology* tab:
+
[source,yaml,subs="+quotes"]
----
upstream:
  backstage:
    appConfig:
      dynamicPlugins:
        frontend:
          backstage-community.plugin-topology:
            entityTabs:
              - mountPoint: entity.page.topology
                path: /topology
                title: Topology
                mountPoint: _<tab_mount_point>_
                priority: -1
----

> . To add a new tab, add a new list item in the `entityTabs` section, and define the `path`, `title`, `mountPoint`, and `priority` values:

Author: ```suggestion
. To add a new tab, add a new list item in the _<plugin_name>_ `entityTabs` section, and define the `path`, `title`, `mountPoint`, and `priority` values:
```

> . To add a new tab, add a new list item in the _<plugin_name>_ `entityTabs` section, and define the `path`, `title`, `mountPoint`, and `priority` values:

Author: ```suggestion
. To add a new tab, add a new list item in the _<plugin_name>_ plugin `entityTabs` section, and define the `path`, `title`, `mountPoint`, and `priority` values:
```

+
[source,yaml,subs="+quotes"]
----
upstream:
  backstage:
    appConfig:
      dynamicPlugins:
        frontend:

>           entity-tabs-customization:

Author: ```suggestion
          _<plugin_name>_:
```
